### PR TITLE
DDF, improve support for Xiaomi Aqara relay controller lumi.relay.c2acn01

### PR DESCRIPTION
**Pull request by Smanar**
Long story, it seem this device have issues
https://github.com/dresden-elektronik/deconz-rest-plugin/issues/6512
https://forum.phoscon.de/t/device-appear-in-deconz-but-not-in-phoscon/3298

Solved by this DDF

### CODE REVIEW

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+          "name": "attr/manufacturername"
+        },
+        {
+          "name": "attr/modelid"
+        },
+        {
+          "name": "attr/name"
+        },
+        {
+          "name": "attr/swversion"
```
Comment (SwoopX) [inline devices/xiaomi/xiaomi_lumi.relay.c2acn01.json:36]: Could you please add the respective content for getting the firmware version here and for the other sensors? You can take the P1 presence sensor as reference here, same idx (0x08)

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+        },
+        {
+          "name": "attr/type"
+        },
+        {
+          "name": "attr/uniqueid"
+        },
+        {
+          "name": "state/alert",
+          "default": "none"
```
Comment (SwoopX) [inline devices/xiaomi/xiaomi_lumi.relay.c2acn01.json:46]: You could remove the default here as it is already included in the generic item

**Review by SwoopX — CHANGES_REQUESTED** [review]

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json (hunk 1 above)
Comment (Smanar) [reply devices/xiaomi/xiaomi_lumi.relay.c2acn01.json]: done

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+        },
+        {
+          "name": "state/on",
+          "refresh.interval": 5
```
[the remark this answers is not in the document]
Comment (Smanar) [reply devices/xiaomi/xiaomi_lumi.relay.c2acn01.json]: done

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+        },
+        {
+          "name": "attr/type"
+        },
+        {
+          "name": "attr/uniqueid"
+        },
+        {
+          "name": "state/alert",
+          "default": "none"
```
[the remark this answers is not in the document]
Comment (Smanar) [reply devices/xiaomi/xiaomi_lumi.relay.c2acn01.json]: done

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+            "cl": "0x000C",
+            "ep": 1,
+            "fn": "zcl"
+          },
+          "parse": {
+            "at": "0x0055",
+            "cl": "0x000C",
+            "ep": 1,
+            "eval": "Item.val = Math.round(Attr.val);"
+          }
```
[the remark this answers is not in the document]
Comment (Smanar) [reply devices/xiaomi/xiaomi_lumi.relay.c2acn01.json]: Heren IDK, I don't the device, but return seem fine and it's the same code than in this DDF https://github.com/dresden-elektronik/deconz-rest-plugin/blob/master/devices/xiaomi/xiaomi_lumi.switch.n0agl1_switch.json#L126

[the remark this answers is not in the document]
Comment (Smanar) [reply devices/xiaomi/xiaomi_lumi.relay.c2acn01.json]: Ha right, I thought the problem was from power but it's consumption the problem.
Have made some search and I realy have no clue where this value is from, Using the idx 95 to test for the moment.

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json (hunk 2 above)
Comment (Smanar) [reply devices/xiaomi/xiaomi_lumi.relay.c2acn01.json]: done

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+          "name": "attr/manufacturername"
+        },
+        {
+          "name": "attr/modelid"
+        },
+        {
+          "name": "attr/name"
+        },
+        {
+          "name": "attr/swversion"
```
[the remark this answers is not in the document]
Comment (Smanar) [reply devices/xiaomi/xiaomi_lumi.relay.c2acn01.json]: done

File: devices/xiaomi/xiaomi_lumi.relay.c2acn01.json [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+        {
+          "name": "attr/uniqueid"
+        },
+        {
+          "name": "state/alert",
+          "default": "none"
+        },
+        {
+          "name": "state/on",
+          "refresh.interval": 5
```
[the remark this answers is not in the document]
Comment (Smanar) [reply devices/xiaomi/xiaomi_lumi.relay.c2acn01.json]: done

**Comment by manup** [comment]
Thanks for the fix, is this one good to merge or wip yet?

**Comment by SwoopX** [comment]
> Thanks for the fix, is this one good to merge or wip yet?

We're not quite there yet. The consumption values does not have the correct unit (must be * 1000) and the power sensor should use the correct cluster (000C) in the uuid, I guess? Not too sure if the latter could interfere with older sensors however. That's probably more a blemish one can live with?

**Comment by monpelaud** [comment]
One thing is sure, currently the device does not work! With this ddf file the consumption value seems uncertain but the device works and fulfills its basic function.

**Comment by Smanar** [comment]
>the power sensor should use the correct cluster (000C) in the uuid, I guess? 

Right, have modified it.

>The consumption values does not have the correct unit (must be * 1000)

Here I realy have no clue, I haven't found how the device send consumption, and the user that make test, haven't enought consumption to test current and consumption.

**Comment by manup** [comment]
Is the PR ready to merge? :)
The v2.21.0-beta is mainly about fixes but I guess this PR falls also into this category.

**Comment by SwoopX** [comment]
> Is the PR ready to merge? :) The v2.21.0-beta is mainly about fixes but I guess this PR falls also into this category.

Well, I'm afraid the consumption value is still incorrect as per my comment above and it has been like that with the legacy code.

**Comment by monpelaud** [comment]
On Silver support is a good idea.

**Comment by Mimiix** [comment]
@Smanar  @manup  Can this be achieved prior the release?

**Comment by Smanar** [comment]
Have set "Silver" Status,
Only the consumption can't be not working, and not possible to solve the problem yet.

**Comment by manup** [comment]
Yes silver status is a good choice here until the consumption is figured out.

**Pull request merged**